src/main/java/gg/archipelago/aprandomizer/events/onDeathLink.java

```diff
@@ -1,48 +1,29 @@
 package gg.archipelago.aprandomizer.events;
 
-import com.mojang.brigadier.exceptions.CommandSyntaxException;
 import gg.archipelago.aprandomizer.APRandomizer;
 import gg.archipelago.aprandomizer.common.DeathLinkDamage;
 import gg.archipelago.aprandomizer.common.Utils.Utils;
 import gg.archipelago.client.events.ArchipelagoEventListener;
-import gg.archipelago.client.events.BouncedEvent;
-import net.minecraft.core.Holder;
-import net.minecraft.core.registries.Registries;
-import net.minecraft.nbt.CompoundTag;
-import net.minecraft.nbt.TagParser;
-import net.minecraft.resources.ResourceKey;
+import gg.archipelago.client.events.DeathLinkEvent;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.world.damagesource.DamageType;
-import net.minecraft.world.entity.Entity;
-import net.minecraft.world.entity.EntityType;
-import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.level.GameRules;
-import net.minecraft.world.phys.Vec3;
-
-import java.util.concurrent.ThreadLocalRandom;
 
 public class onDeathLink {
 
     @ArchipelagoEventListener
-    public static void onBounce(BouncedEvent event) {
-        if(event.tags.contains("DeathLink") && APRandomizer.getAP().getSlotData().deathlink) {
-            if(event.getDouble("time") == APRandomizer.getLastDeathTimestamp())
+    public static void onDeathLink(DeathLinkEvent event) {
+        if(APRandomizer.getAP().getSlotData().deathlink) {
+            if(event.time == APRandomizer.getLastDeathTimestamp())
                 return;
 
-            GameRules.BooleanValue showDeathMessages = APRandomizer.getServer().getGameRules().getRule(GameRules.RULE_SHOWDEATHMESSAGES);
-            boolean showDeaths = showDeathMessages.get();
-            if(showDeaths) {
```

Comment:
The reason for this is for 2 things.
1. Respects the built in game rule. Not showing the death cause if the rule is set to false.
2. Stops the vanilla death message from showing up in favor of our custom one. With lots of players on the server seeing that every player died due to death link damage, on top of our custom death message just spams the chat.

Reply:
Makes sense!